src/xrGame/space_restriction_composition.cpp

```diff
@@ -153,17 +153,11 @@ void CSpaceRestrictionComposition::test_correctness()
     {
         for (const auto& restriction : m_restrictions)
         {
-            // Check if the restrictor has has a border
-            if (restriction->object().m_test_storage.empty())
+            R_ASSERT3_CURE(!restriction->object().m_test_storage.empty(), "Restrictor has no border", restriction->object().name().c_str(),
             {
-                static bool ignoreAlways = false;
-                if (!ignoreAlways)
-                    xrDebug::Fail(ignoreAlways, DEBUG_INFO, "!restriction->object().m_test_storage.empty()",
-                        "Restrictor has no border", *restriction->object().name());
-
-                continue;
-            }
-
+                m_correct = false;
+                break;
```

Comment:
Will this break work correctly as `R_ASSERT3_CURE` wraps everything in a `do { ... } while (false)` loop?

Reply:
What do you mean?

Reply:
`do while` cycle unlike `while` always executes at least once.

Reply:
Ahh.... No, this break won't work correctly.

Reply:
> Ahh.... No, this break won't work correctly.

I believe I first coded this with `R_ASSERT3_CURE` and got a warning from GCC saying that the `continue` is useless or something. That's why I wrote it in that complicated way